### PR TITLE
net: lib: aws_fota: Fix return code for CONNACK and PUBLISH

### PR DESCRIPTION
**Pull request by sigvartmh**
`CONNACK` events and `PUBLISH` events were consumed by the `aws_fota`
library. This resulted in MQTT handlers using it not being able to
correctly handle events as expected.

Signed-off-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>

### CODE REVIEW

File: subsys/net/lib/aws_fota/src/aws_fota.c [hunk 1]
```diff
@@ -412,7 +415,8 @@ int aws_fota_mqtt_evt_handler(struct mqtt_client *const client,
 				return err;
 			}
 		}
-		return 0;
+		/* Don't consume the publish event */
+		return 1;
```
Comment (jtguggedal) [inline subsys/net/lib/aws_fota/src/aws_fota.c]: It looks like the checks earlier (`if (err < 1)`) in the PUBLISH case can return with 0, even if the event should be deferred to another MQTT handler. I think the return value and documentation of `aws_fota_on_publish_evt` might need an update

Comment (sigvartmh) [reply subsys/net/lib/aws_fota/src/aws_fota.c]: Maybe something more like these changes?

**Comment by sigvartmh** [comment]
If you have time could you confirm that this works? I don't have a device with paired/JITP nrfcloud certificates on.

**Comment by jtguggedal** [comment]
@sigvartmh Forgot to mention that I tested on DK, and it looked good. Would be good if someone else tested as well.

**Comment by mbolivar** [comment]
@sigvartmh will you fix the linter issue? That appears to be what is holding up the NRF CI also

**Comment by sigvartmh** [comment]
@mbolivar Yup hopefully fixed it now.

File: subsys/net/lib/aws_fota/src/aws_fota.c [hunk 2]
```diff
@@ -381,23 +384,26 @@ int aws_fota_mqtt_evt_handler(struct mqtt_client *const client,
...
-					      p->message.payload.len);
-		if (err < 1) {
+		err = on_publish_evt(client,
+				     p->message.topic.topic.utf8,
+				     p->message.topic.topic.size,
+				     p->message.payload.len);
+
+		if (err < 0) {
+			return err;
+		} else  if (err == 1) {
```
Comment (rlubos) [inline subsys/net/lib/aws_fota/src/aws_fota.c]: nit: double space before `if`

Comment (sigvartmh) [reply subsys/net/lib/aws_fota/src/aws_fota.c]: Nice, spot didn't see it. Thanks.

**Pull request merged**